packages/lab/src/utils/forwardCallbackProps.ts

```diff
@@ -7,25 +7,27 @@
     forwardCallbackProps(element.props, overrideProps)
   )
  */
-export const forwardCallbackProps = (elementProps: any, overrideProps: any) => {
-  const callbackProps = Object.entries<Function>(elementProps).filter(
-    ([, value]) => typeof value === "function"
-  );
 
-  if (callbackProps.some(([name]) => overrideProps[name] !== undefined)) {
-    return {
-      ...overrideProps,
-      ...callbackProps.reduce((map, [name, fn]) => {
-        if (overrideProps[name] && typeof overrideProps[name] === "function") {
-          map[name] = (...args: any) => {
-            overrideProps[name]?.(...args);
-            fn(...args);
-          };
-        }
-        return map;
-      }, {} as any),
-    };
-  } else {
-    return overrideProps;
-  }
+type Props = Record<string, unknown>;
+
+export const forwardCallbackProps = (
+  ownProps: Props,
+  overrideProps: Props
+): Props => {
+  const props = {
+    ...overrideProps,
+    ...Object.keys(ownProps).reduce<Props>((map, name) => {
+      const ownProp = ownProps[name];
+      const overrideProp = overrideProps[name];
+      if (typeof ownProp === "function" && typeof overrideProp === "function") {
+        map[name] = (...args: unknown[]) => {
+          ownProp(...args);
```

Comment:
call the clients 'ownProp' first. For those overridden callback props where defaultPrevented will cause them to bail, gives the client the ability to handle the event solo.